pkg/localstore/localstore.go

```diff
@@ -670,6 +629,50 @@ func New(path string, baseKey []byte, ss storage.StateStorer, o *Options, logger
 	return db, nil
 }
 
+func (db *DB) safeInit(rootPath, sharkyBasePath string) error {
+	// create if needed
+	path := filepath.Join(rootPath, sharkyDirtyFileName)
+	if _, err := os.Stat(path); errors.Is(err, fs.ErrNotExist) {
+		// missing lock file implies a clean exit then create the file and return
+		return os.WriteFile(path, []byte{}, 0644)
+	}
+
+	db.logger.Warning("localstore sharky .DIRTY file exists: starting recovery due to previous dirty exit")
```

Comment:
Awesome!   You can't imagine how many times I've sat and wondered why my node was taking so long to establish peers.  I've always suspected it was due to an unclean shutdown, but hopefully this log will help me to confirm or deny my theory.

Reply:
so after this change, can you report how often the recovery happens to you if at all ?

Reply:
I'll try to remember to keep an eye on it.   Normally, I've only observed startup delays after a shutdown doesn't go cleanly, either an OOM kill on Ubuntu, or a Control-C on Windows that eventually reports that something terminated with running goroutine.  Or when I get impatient and just hit double Control-C's on windows which instantly kills the process with no hope of any shutdown cleanup!